library/src/blas2/rocblas_syr.hpp

```diff
@@ -0,0 +1,80 @@
+/* ************************************************************************
+ * Copyright 2016-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#pragma once
+#include "handle.h"
+#include "rocblas.h"
+
+template <typename T, typename U>
+__global__ void rocblas_syr_kernel(rocblas_fill uplo,
+                                   rocblas_int  n,
+                                   U            alpha_device_host,
+                                   const T* __restrict__ x,
```

Comment:
Might want to ask Juan if he needs support in the template for a vector of the scalar alpha. We may just want to stay consistent and it anyway. You can see rocblas_scal as an example, just need to add in a stride for alpha and call the new load_scalar helper.

Reply:
I'll ask Juan.

Reply:
I'll add the alpha vector support following the other examples pattern.

Reply:
@mahmoodw  please review again.  Added vector but deferred supporting host vector.